ash/src/extensions/ext/calibrated_timestamps.rs

```diff
@@ -0,0 +1,64 @@
+use crate::prelude::{read_into_uninitialized_vector, VkResult};
+use crate::vk;
+use crate::{Device, Instance};
+use std::ffi::CStr;
+use std::mem;
+
+#[derive(Clone)]
+pub struct CalibratedTimestamps {
+    handle: vk::Device,
+    fp: vk::ExtCalibratedTimestampsFn,
+}
+
+impl CalibratedTimestamps {
+    pub fn new(instance: &Instance, device: &Device) -> Self {
+        let handle = device.handle();
+        let fp = vk::ExtCalibratedTimestampsFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(handle, name.as_ptr()))
+        });
+        Self { handle, fp }
+    }
+
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceCalibrateableTimeDomainsEXT.html>"]
+    pub unsafe fn get_physical_device_calibrateable_time_domains(
+        &self,
+        physical_device: vk::PhysicalDevice,
+    ) -> VkResult<Vec<vk::TimeDomainEXT>> {
+        read_into_uninitialized_vector(|count, data| {
+            self.fp
+                .get_physical_device_calibrateable_time_domains_ext(physical_device, count, data)
```

Comment:
Whoops, we can't actually load this through `get_device_proc_addr`. As we did with at least one other extension, just load it through `get_instance_proc_addr` and take the indirection table hit :(

Reply:
Is the problem that the extension has both an instance-level and a device-level function? Sounds like we should probably handle those separately.